validation/keysin.go

```diff
@@ -0,0 +1,39 @@
+package validation
+
+import "strings"
+
+// KeysInValidator validates the field under validation must contain all the given keys.
+type KeysInValidator struct {
+	BaseValidator
+	Keys []string
+}
+
+// Validate checks the field under validation satisfies this validator's criteria.
+func (v *KeysInValidator) Validate(ctx *Context) bool {
+	obj, ok := ctx.Value.(map[string]interface{})
+	if !ok {
+		return false
+	}
+
+	for _, key := range v.Keys {
+		if _, ok := obj[key]; !ok {
+			return false
+		}
+	}
```

Comment:
This validator should check that the object doesn't contain a key that is not allowed. Here, you are checking that the object contains all the keys, which is different.